### PR TITLE
group-idle role on idle members

### LINKED ISSUE #1422: Add group-idle role
# What to do?
- [x] group-idle should be applied to the members who become idle
- [x] group-idle should be removed when a member is not idle

### PR DESCRIPTION
**Pull request by lakshayman**
closes #1422

### CODE REVIEW

File: models/discordactions.js [hunk 1]
```diff
@@ -47,6 +47,33 @@ const getAllGroupRoles = async () => {
   }
 };
 
+/**
+ *
+ * @param roleData { Object }: Data of the new role
```
Comment (iamitprakash) [inline models/discordactions.js:52]: roleData is never used

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: models/userStatus.js [hunk 2]
```diff
@@ -17,7 +17,65 @@ const { TASK_STATUS } = require("../constants/tasks");
 const userStatusModel = firestore.collection("usersStatus");
 const tasksModel = firestore.collection("tasks");
 const { userState } = require("../constants/userStatus");
+const { getGroupRole, addGroupRoleToMember, removeGroupRoleFromMember } = require("./discordactions");
 const usersCollection = firestore.collection("users");
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+const jwt = require("jsonwebtoken");
+
+const removeGroupRoleFromDiscordUser = async ({ userId, roleName }) => {
```
Comment (iamitprakash) [inline models/userStatus.js:25]: roleid property of the dataForDiscord object is not being set correctly.

The removeGroupRoleFromDiscordUser function takes a userId and roleName parameter and removes the role with the given name from the user's Discord account. The function first calls the getGroupRole function to get the role data, and then calls the removeGroupRoleFromMember function to remove the role from the user's Discord guild. Finally, the function calls the fetch function to make a DELETE request to the Discord API to remove the role from the user's Discord account.

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: models/userStatus.js (hunk 2 above)
Comment (lakshayman) [reply models/userStatus.js:26]: First, it removes the role data from our firestore database.
Yes, getGroupRole is for getting the ID of the role.
removeGroupRoleFromMember is for removing the role from our firestore database.
fetch call is for deleting it from the user on Discord.

File: models/discordactions.js (hunk 1 above)
Comment (lakshayman) [reply models/discordactions.js]: resolved in latest commit

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**